debian/schooltool-server.1
.TH schooltool-server 1
.SH NAME
schooltool-server \- starts the SchoolTool server
.SH SYNOPSIS
.B schooltool-server
.I "[options]"
.SH DESCRIPTION
.B schooltool-server
is the SchoolTool server program.
.SH OPTIONS
.TP
.B \-h, \-\-help
Show summary of options
.TP
.B \-c FILENAME, \-\-config=FILENAME
Use the given configuration file.
.SH FILES
.TP
.B /etc/schooltool.conf
The default server configuration file
.SH "SEE ALSO"
schooltool-client(1), wxschooltool(1).


CHANGES.txt
=======
CHANGES
=======

2.6.4 (unreleased)
------------------

- Support CKEditor 4.x (https://bugs.launchpad.net/bugs/1309133)
- Delete schedules that reference a deleted timetable (https://bugs.launchpad.net/bugs/1280528)
- Improve tab style to make active tab more clear (https://launchpad.net/bugs/1023378)
- Timetables copied from active year retain old start/end dates (https://launchpad.net/bugs/1005962)
- Updated translations


2.6.3 (2014-04-12)
------------------

New features
++++++++++++

- Switch to CKEditor (https://bugs.launchpad.net/bugs/485898)
- Celery 3 support (https://bugs.launchpad.net/bugs/1276384)
- Latest translations from Launchpad

Bugfixes
++++++++

- Fixed exception dates format in calendar widget of recurring events (https://launchpad.net/bugs/372889)
- Fixed export of deleted timetables (https://launchpad.net/bugs/1281335)
- Updated copyright (https://launchpad.net/bugs/1289277)
- Fixed term/section navigation in a different school year (https://launchpad.net/bugs/1281050)


2.6.2 (2014-02-15)
------------------

- Fix upgrade from a fresh install of version 2.6.0 (https://launchpad.net/bugs/1258529)
- Fix delete year/term if it contains non-ascii chars.
- Updated translations


2.6.1 (2013-11-21)
------------------

- Synchronize timezone set in preferences with timetables and schedules (https://launchpad.net/bugs/1239468)
- Set initial timezone from ``/etc/timezone`` (https://launchpad.net/bugs/1030211)
- Add SchoolTool versions to the Server page.
- Person cannot see his own preferences (https://launchpad.net/bugs/1247825)
- Do not escape extra spaces in comments (https://launchpad.net/bugs/1241022)
- Fix error when trying to replace existing school logo (https://launchpad.net/bugs/1250733)
- Restore the warning about unsaved changes in gradebook, broken after adding
  autocompletion (https://launchpad.net/bugs/1253070)
- Updated translations


2.6.0 (2013-10-10)
------------------

- Don't reverse name order (https://launchpad.net/bugs/1201887)
- Add "Name sorting" option to Server settings, display the selected name
  column first in person tables.
- Person search does not find by full name (https://launchpad.net/bugs/894798)
- Change manager's full name to "Default Manager" (https://launchpad.net/bugs/1227220)
- Fix timetable wizard for weekly timetables with different periods order.
- Fix crash when displaying timetables with only one day (https://launchpad.net/bugs/1235150)
- Fix navigation inside the School tab if a non-active school year is selected,
  or school years title contains accented characters (https://launchpad.net/bugs/1234690)
- Setting multiple languages in schooltool.conf breaks remote reports (https://launchpad.net/bugs/1229013)
- Improvements to PDF grid and table style.
- Extra info in traceback page and submit to launchpad button
- Updated translations, removed unsupported ones (less than 25% translated).


2.5.3 (2013-09-17)
------------------

Bugfixes
++++++++

- Fix integer demographic fields handling in importer (https://launchpad.net/bugs/1211613)
- Add Activate school year link in School tab under Customize (https://launchpad.net/bugs/1216992 )
- Fix links and Done buttons inside School tab when a different school year is selected
- Add manager to Site Managers and Clerks groups (https://launchpad.net/bugs/1216915)
- Access setting: allow instructor to add students to their section (https://launchpad.net/bugs/637759)
- Access setting: allow instructor to modify their section (https://launchpad.net/bugs/683313)
- Access setting: instructors can schedule their own sections (https://launchpad.net/bugs/604643)
- Display contact information of a person in profile.
- Updated translations

Internals
+++++++++

- Add ability to disable views in customized packages (https://launchpad.net/bugs/716346)


2.5.2 (2013-08-22)
------------------

- Autocompletion in gradebook for discrete score systems.
- TAB key in gradebook moves to the next cell.
- Further updates to permissions.
- Clerks and administrators now have appropriate access to parts of the School.
- Update GPL license and file headers from gnu.org (https://launchpad.net/bugs/1211145)
- Fix failures when creating timetables with rotating periods and weekly time slots (https://launchpad.net/bugs/1214131)
- Updated translations


2.5.1 (2013-08-01)
------------------

- Differentiate permissions of administrators, clerks, and managers.
- Scripts and styles for the new attendance features.
- Updated translations


2.5.0 (2013-06-03)
------------------

Reports
+++++++

- Background XLS import (disabled)
- Background XLS export
- Background report generation
- Multiple reports in a .zip archive
- Messages about finished background tasks
- Added resource library for filtered/chart reports
- Removed Person XML reports

Server
++++++

- There are now 4 processes, controlled by ``supervisor``: ``schooltool``, ``zeo``,
  ``celery_report``, and ``redis``.
- New ``make`` targets: start, stop, restart, rerun, status.
- You have to install ``redis-server`` package on the system. Other new
  dependencies are Python packages.
- Instances have to be updated to run this version. If you run from a sandbox,
  do ``bin/make-schooltool-instance instance``.


2.4.1 (2013-05-30)
------------------

Reports
+++++++

- Fixed report generation to not do network requests (https://launchpad.net/bugs/1185418).
- Fixed ratio of school logo.
- Fixed rendering of "&" in headers.

Bugfixes
++++++++

- Warn about unsaved changes when leaving gradebook (https://launchpad.net/bugs/907999)
- Comment dialog submits the gradebook

Language
++++++++

- Replaced Filter by... labels
- Updated Italian, Greek, Portuguese, and other translations.


2.4.0 (2013-04-19)
------------------

- Fix crash on startup in 2.4b1 release
- Fix PDF reports crashing with newer z3c.rml (https://launchpad.net/bugs/1169723)
- Remove dependency on ``zope.app.server``
- Remove or deprecate unused schooltool.conf options

schooltool/schooltool-server.py
#!/usr/bin/env python
"""
A script to start the schooltool server from the source directory.
"""

import sys
import os.path

if sys.version_info < (2, 3):
    print >> sys.stderr, '%s: need Python 2.3 or later.' % sys.argv[0]
    print >> sys.stderr, 'Your python is %s' % sys.version
    sys.exit(1)

# find the schooltool config file if it exists
dirname = os.path.dirname(__file__)
config_file = os.path.join(dirname, 'schooltool.conf')
if not os.path.exists(config_file):
    config_file = os.path.join(dirname, 'schooltool.conf.in')

# Change the default config file name by prepending a command-line argument
sys.argv.insert(1, '--config=' + config_file)

import os
basedir = os.path.abspath(os.path.dirname(sys.argv[0]))
sys.path.insert(0, os.path.join(basedir, 'src'))
sys.path.insert(0, os.path.join(basedir, 'Zope3', 'src'))

import schooltool.main
schooltool.main.StandaloneServer().main()
